 Exception Processing is important to find exceptions that causes the runtime error. As runtime errors Halt the program execution when exception occurs.

Exception Processing is used in Various Applications of which few examples are:

Checking Appropriate use of input in an application
Checking for Arithmetic exceptions in mathematical executions
Checking File I/O exceptions during File handling



If Exceptions are not handled flow of program will be broken during the run time which might lead to a abnormal termination of the program. 
Inshort inability of program to handle exceptions will result in crashing of program

 Python provides try and except statements for recovering from an exception in your script.


raise and assert are two methods that can be used to trigger manual exceptions in your script.

raise method triggers an exception if condition provided to it turns out to be True.
assert will let the program to continue execution if condition provided to it turns out to be True else exception will be raised

# Example of raise
x = 10
raise Exception(f'X Value Should not exceed 5 The Provided Value of X is {x}')


# Example of assert
assert(2==4), "2 is not equal to 4"

Python Provides else and finally blocks for specifying actions to be executed at termination time, regardless of whether an exceptions exists or not.